Add Flatpak manifest

t2b3 opened the pull request:
This Pull Request adds the option to build a FlatPak and adds some conventions to fulfill the requirements of the Flatpak specification.

In order to successfully build and install it, wxWidgets has to be built and linked statically into TreeSheets but we can benefit from the newly added option to do exactly that.

The stuff related only to Linux is moved into `linux`, i.e. FreeDesktop related stuff.

This can be a step forward to move away from own pre-built binaries that create a lot of issues.

aardappel commented on TS/org.aardappel.TreeSheets.metainfo.xml:
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Do all these metadata files have to sit in `TS` ? `TS` is also the distribution folder for other platforms, that clearly don't need this file.

t2b3 replied:
I put the Linux-only stuff into a separate folder.

aardappel commented:
@t2b3 and yes, if you want to maintain it, feel free to make a flatpak available

t2b3 commented:
Hello Wouter
Thanks for your feedback. I removed the submodule and added instructions in the Flatpak manifest to extract the wxWidgets source into the folder `lib/wxWidgets`. In addition to that I moved all the Linux-only stuff out of `TS` and to the folder `linux` and changed the CMakefile to take the changes into account.

t2b3 commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "lib/wxWidgets"]
+	path = lib/wxWidgets
```
Reverted that changed.

t2b3 commented:
Hello Wouter
All checked passed. This PR would be good to go from my side.
I would like to ask you to review it and if it is ok, to merge it. Many thanks in advance!
Best regards
Tobias

t2b3 commented:
Thanks for the merge :-)
